File: JavaPrograms/Loopingstatement/src/loopingstatement/ArmstrongNumber.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package loopingstatement;

/**
 *program name : To find a Armstrong number
 * @author harid
 */
import java.util.Scanner;
public class ArmstrongNumber {
public static void main(String[] args) {
Scanner input=new Scanner(System.in);
System.out.println("enter the number");
int num=input.nextInt();
int num1,rem,cube,sum=0;
num1=num;
while(num>0)
{
    cube=1;
    rem=num%10;
    for(int i=1;i<=num;i++)
    {
        cube=rem*rem*rem;
    }  
    sum=sum+cube;
    num=num/10;
    }
    System.out.println(sum);
if(sum==num1)
        System.out.println("the given number is a armstrong number");
else
        System.out.println("the given number is not a armstrong number");
}
}

File: CPP_Programs/armstrong_number.cpp
//program name : armstrong number
#include<iostream>
using namespace std;
class arm
{
	public:
	int a,b,i,c,plus=0,d;
	void sum()
	{
	cout<<"enter the number";
	cin>>a;
	d=a;
	for(i=1;i<=a;i++)
	{
		b=a%10;
		c=b*b*b;
		plus=plus+c;
		a=a/10;
	}
	cout<<plus;
}
};
class check:public arm
{

public:
	void num()
	{
	if(d==plus)
	{
	cout<<"\nthe given number is a armstrong number";
	}
	else
	{
		cout<<"\nthe given number is not a armstrong number";
	}
}};
int main()
{
	check d;
	d.sum();
	d.num();
}
